### PR TITLE
Fixing 512 byte issue and accounting for tsize and blksize options in ack packets in TFTP

### PR DESCRIPTION
**Pull request by nadazeini**

### CODE REVIEW

**Comment by hibozahara** [comment]
Looks good - approved!

**Comment by breed** [comment]
it looks like the test for 512 byte sized files is missing.

**Review by breed — CHANGES_REQUESTED** [review]
please add the test for files that are multiples of 512-bytes

**Comment by nadazeini** [comment]
Done!

File: tftp/test_tftp.py [hunk 1]
```diff
@@ -166,11 +167,165 @@ def test_bootcode(self):
         except Exception as e:
             self.fail(e)
         finally:
+            os.remove("./bootcodeTest.bin")
+            tftp_thread.stop()
+
+    def test_512(self):
```
Comment (breed) [inline tftp/test_tftp.py]: it looks like you have duplicated code here. can you pull that out into a function? (duplicated code is hard to maintain!)

**Review by breed — CHANGES_REQUESTED** [review]
looks good, just a small restructure requested.

File: tftp/test_tftp.py (hunk 1 above)
Comment (nadazeini) [reply tftp/test_tftp.py]: I modified it to use the same test with different parameters

File: tftp/test_tftp.py [hunk 2]
```diff
@@ -31,14 +30,26 @@ class tftp_tests(unittest.TestCase):
...
-        tftp_thread = tftp.TFTPServer(data_dir, tftp_port, ip)
+    def test(self):
+        tftp_thread = tftp.TFTPServer(self.data_dir, self.tftp_port, ip)
         tftp_thread.start()
-        filename = "start.elf"
         try:
-            with socket.socket(family=socket.AF_INET, type=socket.SOCK_DGRAM) as sd, open(f'startTest.elf',
+            if self.file_size != -1:
+                # generate file of 512 bytes
+                print("here")
```
Comment (breed) [inline tftp/test_tftp.py]: remove this debug statement :)

**Review by breed — CHANGES_REQUESTED** [review]
wow! super fancy! just need to remove a debug statement

**Review by breed — APPROVED** [review]
nice work

**Pull request merged**